### PR TITLE
Integrate listener and fulltext adapter.

### PR DESCRIPTION
**Pull request by bright-starry-sky**

### CODE REVIEW

File: src/kvstore/plugins/elasticsearch/ESListener.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        return false;
+    }
+    auto raw = encodeAppliedId(lastId, lastTerm, lastApplyLogId);
+    ssize_t written = write(fd, raw.c_str(), raw.size());
+    if (written != (ssize_t)raw.size()) {
+        VLOG(3) << idStr_ << "bytesWritten:" << written << ", expected:" << raw.size()
+                << ", error:" << strerror(errno);
+        return false;
+    }
+    return true;
```
Comment (critical27) [inline src/kvstore/plugins/elasticsearch/ESListener.cpp:87]: Do we need to close(fd)?

**Comment by bright-starry-sky** [comment]
Done. ready to review.

File: src/kvstore/plugins/elasticsearch/ESListener.cpp [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    }
+    return appendDocs(items, schema.get(), reader.get(), tagId, false);
+}
+
+bool ESListener::appendDocs(std::vector<DocItem>& items,
+                            const meta::SchemaProviderIf* schema,
+                            RowReader* reader,
+                            int32_t schemaId,
+                            bool isEdge) const {
+    auto spaceName = getSpaceName();
```
Comment (critical27) [inline src/kvstore/plugins/elasticsearch/ESListener.cpp:179]: We don't need to get space name for each DocItem?

Comment (bright-starry-sky) [reply src/kvstore/plugins/elasticsearch/ESListener.cpp:179]: > We don't need to get space name for each DocItem?

In this clip. All DocItems have the same space name, so we don't need to get space name for each DocItem. WDYT?

Comment (critical27) [reply src/kvstore/plugins/elasticsearch/ESListener.cpp:179]: If so, we could add a interface in `Listener` to init.

File: src/kvstore/plugins/elasticsearch/ESListener.cpp [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+namespace nebula {
+namespace kvstore {
+bool ESListener::apply(const std::vector<KV>& data) {
+    std::vector<nebula::plugin::DocItem> docItems;
+    auto vIdLen = getVidLen();
+    if (!vIdLen.ok()) {
+        return false;
+    }
+    for (const auto& kv : data) {
```
Comment (critical27) [inline src/kvstore/plugins/elasticsearch/ESListener.cpp:21]: Maybe we need to split into batch if necessary?

File: src/kvstore/plugins/elasticsearch/ESListener.cpp (hunk 2 above)
Comment (bright-starry-sky) [reply src/kvstore/plugins/elasticsearch/ESListener.cpp:190]: > If so, we could add a interface in `Listener` to init.

Aha, I also think it's a good idea.

File: src/kvstore/plugins/elasticsearch/ESListener.cpp (hunk 3 above)
Comment (bright-starry-sky) [reply src/kvstore/plugins/elasticsearch/ESListener.cpp]: > Maybe we need to split into batch if necessary?

Yeah, solved.

**Comment by bright-starry-sky** [comment]
Refer to https://github.com/vesoft-inc/nebula-common/pull/260

**Comment by bright-starry-sky** [comment]
Disabled es_bulk_test for regression test.  because this test case is taking too long. changed it from test to executable example.

**Review by critical27 — COMMENTED** [review]
Generally LGTM

**Comment by bright-starry-sky** [comment]
move vidLen, spaceName, esClients initialize logic to `init` method

**Review by panda-sheep — DISMISSED** [review]
Good job

**Comment by bright-starry-sky** [comment]
OOM leads to CI(ubuntu1804, gcc-9.2) failed. The last one was successful. Please Ignore it.

**Review by panda-sheep — APPROVED** [review]
LGTM

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**